src/north/CsvToHttp/CsvToHttp.class.js

```diff
@@ -0,0 +1,241 @@
+const Papa = require('papaparse')
+const fs = require('fs')
+
+const ApiHandler = require('../ApiHandler.class')
+
+/**
+ * Class CsvToHttp - convert a csv file into http request such as POST/PUT/PACTH
+ */
+class CsvToHttp extends ApiHandler {
+  /**
+   * Constructor for CsvToHttp
+   * @constructor
+   * @param {Object} applicationParameters - The application parameters
+   * @param {Engine} engine - The Engine
+   * @return {void}
+   */
+  constructor(applicationParameters, engine) {
+    super(applicationParameters, engine)
+
+    const { applicativeHostUrl, requestMethod, authentication, csvDelimiter, mapping, proxy = null } = applicationParameters.CsvToHttp
+
+    // Creation of an entity request with is composed of all necessaries informations
+    this.request = {}
+    this.request.host = applicativeHostUrl
+    this.request.method = requestMethod
+    this.request.authentication = authentication
+
+    this.mapping = mapping
+    this.mapping.csvDelimiter = csvDelimiter
+    this.proxy = this.getProxy(proxy)
+
+    this.httpBody = []
+    this.canHandleFiles = true
+  }
+
+  /**
+   * Send the file.
+   * @param {String} filePath - The path of the file
+   * @return {Promise} - The send status
+   */
+  async handleFile(filePath) {
+    // Verify that the file receive is a csv one
+    const regexExp = RegExp('.csv$')
+
+    if (!regexExp.test(filePath)) {
+      this.logger.error(`Invalid file format, file (${filePath} skipped`)
+      return ApiHandler.STATUS.LOGIC_ERROR
+    }
+
+    await this.parseAndMapp(filePath)
+
+    if (this.httpBody !== []) {
+      await this.sendData()
+      this.logger.silly('Request send successfully')
+    }
+    return ApiHandler.STATUS.SUCCESS
+  }
+
+  /**
+   * @param {string} filePath - PAth of the file to parse
+   * @return {Promise} - Promise
+   */
+  parseAndMapp(filePath) {
+    return new Promise((resolve, reject) => {
+      try {
+        // Initialize array wich will be filled at each step
+        const csvDataParsed = []
+        const csvFile = fs.createReadStream(filePath)
+        // Parse the csv file into a json object
+        Papa.parse(csvFile, {
+          delimiter: this.mapping.csvDelimiter,
+          header: true,
+          dynamicTyping: true,
+          error: (error) => {
+            this.logger.error(error)
+            return ApiHandler.STATUS.LOGIC_ERROR
+          },
+          step: (step) => {
+            csvDataParsed.push(step.data)
+          },
+          complete: () => {
+            // The csv parsing is a success so we begin to map it
+            // Initialize the body to an empty array
+            this.httpBody = []
+            this.convertToHttpBody(csvDataParsed)
+          },
+        })
+        resolve()
```

Comment:
I don't think the resolve of the promise should be used here, but rather once the complete callback of papaparser is called

Reply:
`convertToHttpBody` is not anymore a async function (it was one with csvtojson parser)

Reply:
Do you prefer that `parseAndMapp` don't return a Promise?

Reply:
Papa.parse seems async to me : https://www.papaparse.com/docs#local-files

If you remove the promise or keep the resolve where it is now, what will happen is that the resolve() can fire before the csv is completely parse, so even before the complete() of Papa.parse is called. In this case you won't have any data to send

Reply:
I see the issue but not sure how to deals with...